### PR TITLE
fix/feature (bbb-web): Add new join param `errorRedirectUrl` and fix to respect redirect

### LINKED ISSUE #11241: Reaching the maximum number of participants in join API redirects when "redirect=false"
**Describe the bug**

**To Reproduce**
1. Join a meeting that reached the maximum number of participants, with "redirect=false"
2. Get redirected to the logout page.

**Expected behavior**
An XML output should be displayed with no redirection.

**Actual behavior**
The participant gest redirected to the default logout page.

**Code**
https://github.com/bigbluebutton/bigbluebutton/blob/v2.2.x-release/bigbluebutton-web/grails-app/controllers/org/bigbluebutton/web/controllers/ApiController.groovy#L467-L471

https://github.com/bigbluebutton/bigbluebutton/blob/v2.2.x-release/bigbluebutton-web/grails-app/controllers/org/bigbluebutton/web/controllers/ApiController.groovy#L53

### LINKED ISSUE #12824: Redirect on maxParticipantsReached
**Describe the bug**
If the user joins a meeting using the join api with redirect true and the max. number of users is reached, the user is getting redirected to https://HOSTNAME/?errors=[%7B%22message%22:%22The%20number%20of%20participants%20allowed%20for%20this%20meeting%20has%20been%20reached.%22,%22key%22:%22maxParticipantsReached%22%7D]

**To Reproduce**
1. Set max participants
2. Join users until max is reached
3. Join with an other user
4. See bbb default api page with the displayed error

**Expected behavior**
It would be great, if you could either set an error redirect url on the api join parameters or use the logout url for redirection.

Using the logout isn't working, because the param is not set for join api calls https://github.com/bigbluebutton/bigbluebutton/blob/1abedcc698b16a315b5c2b6b618d41b027c20560/bigbluebutton-web/grails-app/controllers/org/bigbluebutton/web/controllers/ApiController.groovy#L1404

**Actual behavior**
User is getting redirected to bbb api demo page

**Screenshots**
image

**Desktop (please complete the following information):**
 - OS: Windows
 - Browser: Chrome
 - Version: 91.0.4472.124

**Additional context**

### PR DESCRIPTION
**Pull request by paultrudel**
### What does this PR do?

The `join` API did not respect the `redirect` parameter and would continue to redirect users on errors even if `redirect` was set to false. Also added support for a new parameter on `join` API calls called `redirectUrl` that allows a user to either provide a custom URL to redirect users to, or it can be set to `logout` to redirect the user to the meeting's logout URL.

### Closes Issue(s)
Closes #11241
Closes #12824

### CODE REVIEW

**Comment by antobinary** [comment]
re-tagging for 2.7

File: bigbluebutton-web/grails-app/controllers/org/bigbluebutton/web/controllers/ApiController.groovy [hunk 1]
```diff
@@ -207,11 +207,23 @@ class ApiController {
 
     HashMap<String, String> roles = new HashMap<String, String>();
 
-    roles.put("moderator", ROLE_MODERATOR);
-    roles.put("viewer", ROLE_ATTENDEE);
+    roles.put("moderator", ROLE_MODERATOR)
+    roles.put("viewer", ROLE_ATTENDEE)
+
+    //check if exists the param redirect
+    boolean redirectClient = true
```
Comment (gustavotrott) [inline bigbluebutton-web/grails-app/controllers/org/bigbluebutton/web/controllers/ApiController.groovy:214]:
```suggestion
    boolean redirectClient = REDIRECT_RESPONSE
```

File: bigbluebutton-web/grails-app/controllers/org/bigbluebutton/web/controllers/ApiController.groovy [hunk 2]
```diff
@@ -207,11 +207,23 @@ class ApiController {
...
+    String clientURL = paramsProcessorUtil.getDefaultHTML5ClientUrl();
+
+    if (!StringUtils.isEmpty(params.redirect)) {
+      try {
+        redirectClient = Boolean.parseBoolean(params.redirect);
+      } catch (Exception ignored) {
+        redirectClient = true
+      }
```
Comment (gustavotrott) [inline bigbluebutton-web/grails-app/controllers/org/bigbluebutton/web/controllers/ApiController.groovy:222]:
```suggestion
      } catch (Exception ignored) {}
```

**Comment by gustavotrott** [comment]
I think the option `redirectUrl=logout` is not necessary.
Because this is the default behavior already.

image

And as we can see in the print it also override the default `defaultServerUrl` when the `/join` has `params.logoutURL`. I suggest to remove the option `redirectUrl=logout` and keep the previous behavior that is already what we want.

**Comment by gustavotrott** [comment]
hey @paultrudel , the code is alright!

are you going to update the Docs in this PR? or in a subsequent?

**Comment by paultrudel** [comment]
> hey @paultrudel , the code is alright!
> 
> are you going to update the Docs in this PR? or in a subsequent?

I am going to open another PR to update the docs.

**Review by gustavotrott — APPROVED** [review]
LGTM!

**Pull request merged**